core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -289,7 +289,9 @@ class LogCleaner(val config: CleanerConfig,
           }
       }
       if (!cleaned)
-        backOffWaitLatch.await(config.backOffMs, TimeUnit.MILLISECONDS)
+        if (backOffWaitLatch.await(config.backOffMs, TimeUnit.MILLISECONDS)) {
```

Comment:
A bit orthogonal to the intent of this patch, but I feel there is some redundance between the `backOffWaitLatch` we have here and the `isRunning` flag that we have in `ShutdownableThread`. Here's a suggestion:

1. Replace the `isRunning` flag with a latch called `shutdownInitiated`.
2. Change the name of `shutdownLatch` to `shutdownComplete`.
3. Add a method in `ShutdownableThread` to support thread "pausing" which just awaits the `shutdownInitiated` latch.  
4. Remove `backOffWaitLatch` and instead use the new pause API.

What do you think?

Reply:
We appear to have at least one other use of `ShutdownableThread` where a pause or backoff capability would be useful: `ControllerChannelManager.RequestSendThread` just uses `sleep` to backoff.

Reply:
@hachikuji yes, I think your suggestion is an improvement. We should probably review the various subclasses to see if we actually need `interrupt = true` after that change.

Reply:
@hachikuji  currently ShutdownableThread.initiateShutdown method interrupts the thread  and return true on first invocation. there are few subclasses which uses the return value. we need to implement same behavior in your suggested approach. Not sure we can achieve this wth latch alone.

Reply:
@omkreddy Sure, I think @ijuma was just suggesting that we check usages to see if it's possible. Perhaps it isn't, but I think the rest of the suggested refactor still makes sense.

Reply:
@hachikuji  Updated the PR. Pls take a look